src/code.cloudfoundry.org/cf-tcp-router/configurer/haproxy/config_marshaller_test.go

```diff
@@ -350,4 +358,155 @@ backend backend_80
+						InstanceId:      "host-88-instance-id",
+						ExternalPort:    1025,
+
+						SniHostname:          Ptr("rmq1.sys.tas.foobar.com"),
+						TerminateFrontendTLS: Ptr(true),
+					}},
+					{TcpMappingEntity: apimodels.TcpMappingEntity{
+						RouterGroupGuid: "foobar",
+						HostPort:        8889,
+						HostIP:          "88.0.0.37",
+						InstanceId:      "host-89-instance-id",
+						ExternalPort:    1025,
+
+						SniHostname:          Ptr("rmq2.sys.tas.foobar.com"),
+						TerminateFrontendTLS: Ptr(true),
+					}},
+				})
+				Expect(actual).To(Equal(`
+frontend frontend_1025
+  mode tcp
+  bind :1025 ssl crt /fake/path/to/certs/
+  tcp-request inspect-delay 5s
+  tcp-request content accept if { req.ssl_hello_type gt 0 }
+  use_backend backend_1025_rmq1.sys.tas.foobar.com if { ssl_fc_sni rmq1.sys.tas.foobar.com }
+  use_backend backend_1025_rmq2.sys.tas.foobar.com if { ssl_fc_sni rmq2.sys.tas.foobar.com }
+
+backend backend_1025_rmq1.sys.tas.foobar.com
+  mode tcp
+  server server_88.0.0.36_8888 88.0.0.36:8888
+
+backend backend_1025_rmq2.sys.tas.foobar.com
+  mode tcp
+  server server_88.0.0.37_8889 88.0.0.37:8889
+`))
+
+			})
+		})
+		It("for a route with TLS terminating at the router and starting another TLS session with backend", func() {
+			actual := getHAProxyStanza([]apimodels.TcpRouteMapping{
+				{TcpMappingEntity: apimodels.TcpMappingEntity{
+					RouterGroupGuid: "foobar",
+					HostPort:        8888,
+					HostIP:          "88.0.0.36",
+					InstanceId:      "host-88-instance-id",
+					ExternalPort:    1025,
+
+					HostTLSPort:          8888,
+					SniHostname:          Ptr("rmq1.sys.tas.foobar.com"),
+					TerminateFrontendTLS: Ptr(true),
+				}},
+				{TcpMappingEntity: apimodels.TcpMappingEntity{
+					RouterGroupGuid: "foobar",
+					HostPort:        8889,
+					HostIP:          "88.0.0.37",
+					InstanceId:      "host-89-instance-id",
+					ExternalPort:    1025,
+
+					HostTLSPort:          8889,
+					SniHostname:          Ptr("rmq2.sys.tas.foobar.com"),
+					TerminateFrontendTLS: Ptr(true),
+				}},
+				// TODO add bind ssl crt
+			})
+			Expect(actual).To(Equal(`
+frontend frontend_1025
+  mode tcp
+  bind :1025 ssl crt /fake/path/to/certs/
+  tcp-request inspect-delay 5s
+  tcp-request content accept if { req.ssl_hello_type gt 0 }
+  use_backend backend_1025_rmq1.sys.tas.foobar.com if { ssl_fc_sni rmq1.sys.tas.foobar.com }
+  use_backend backend_1025_rmq2.sys.tas.foobar.com if { ssl_fc_sni rmq2.sys.tas.foobar.com }
+
+backend backend_1025_rmq1.sys.tas.foobar.com
+  mode tcp
+  server server_88.0.0.36_8888 88.0.0.36:8888 ssl verify required verifyhost host-88-instance-id ca-file /fake/path/to/ca.pem
+
+backend backend_1025_rmq2.sys.tas.foobar.com
+  mode tcp
+  server server_88.0.0.37_8889 88.0.0.37:8889 ssl verify required verifyhost host-89-instance-id ca-file /fake/path/to/ca.pem
+`))
+
+		})
+	})
 })
+
+func getHAProxyStanza(routeMappings []apimodels.TcpRouteMapping) string {
```

Comment:
By creating this function your tests are different from the rest of the tests in this file. Either refactor all tests, or make your tests follow the pattern of the other tests in the file.